lib/plugins/aws/package/compile/events/sns.js

```diff
@@ -54,228 +54,195 @@ class AwsCompileSNSEvents {
   }
 
   compileSNSEvents() {
-    const template = this.serverless.service.provider.compiledCloudFormationTemplate;
+    const cfTemplate = this.serverless.service.provider.compiledCloudFormationTemplate;
 
     this.serverless.service.getAllFunctions().forEach((functionName) => {
       const functionObj = this.serverless.service.getFunction(functionName);
 
       if (functionObj.events) {
         functionObj.events.forEach((event) => {
-          if (event.sns) {
-            let topicArn;
-            let topicName;
-            let region;
-            let redrivePolicy;
-            if (typeof event.sns === 'object') {
-              if (event.sns.arn) {
-                topicArn = event.sns.arn;
-                if (typeof topicArn === 'string') {
-                  // NOTE: we need to process the topic ARN that way due to lacking
-                  // support of regex lookbehind in Node.js 6.
-                  // Once Node.js 6 support is dropped we can change this to:
-                  // `const splitArn = topicArn.split(/(?<!:):(?!:)/);`
-                  const splitArn = topicArn
-                    .replace(/::/g, '@@')
-                    .split(':')
-                    .map((s) => s.replace(/@@/g, '::'));
-                  topicName = splitArn[splitArn.length - 1];
-                  if (splitArn[3] !== this.options.region) {
-                    region = splitArn[3];
-                  }
-                }
-                topicName = event.sns.topicName || topicName;
-                // Only true when providing CFN intrinsic function (not string) to arn property without topicName property
-                if (!topicName) {
-                  throw new ServerlessError(
-                    'Missing "sns.topicName" setting (it needs to be provided if "sns.arn" is not provided as plain ARN string)',
-                    'SNS_MISSING_TOPIC_NAME'
-                  );
+          if (!event.sns) {
+            return;
+          }
+
+          let topicArn;
+          let topicName;
+          let region;
+
+          if (typeof event.sns === 'object') {
+            if (event.sns.arn) {
+              topicArn = event.sns.arn;
+              if (typeof topicArn === 'string') {
+                const splitArn = topicArn.split(/(?<!:):(?!:)/);
```

Comment:
Can we submit this change with other PR (ideally if each PR focuses on one thing)